src/test/java/pl/joegreen/lambdaFromString/LambdaFactoryTest.java

```diff
@@ -1,240 +1,279 @@
 import pl.joegreen.lambdaFromString.dummy.CustomInterface;
 import pl.joegreen.lambdaFromString.dummy.CustomInterfaceUsingInnerClass;
 
 import javax.tools.JavaCompiler;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
 import java.lang.reflect.Method;
 import java.math.BigDecimal;
 import java.net.URL;
 import java.net.URLClassLoader;
 import java.util.AbstractMap.SimpleEntry;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 import java.util.function.IntBinaryOperator;
 import java.util.function.Supplier;
 import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+class LambdaFactoryTest {
+
+	public static final String INCORRECT_CODE = "()->a";
+
+	static Stream<Arguments> jdkAndEclipse() {
+		int javaVersion = DefaultClassFactory.getJavaVersion();
+
+		if (javaVersion <= 8) {
+			return Stream.of(Arguments.of(JavaCompilerProvider.getJdkJavaCompiler().get()),
+					Arguments.of(new EclipseCompiler()));
+		}
+
+		return Stream.of(Arguments.of(JavaCompilerProvider.getJdkJavaCompiler().get()));
+	}
+
+	@ParameterizedTest
+	@MethodSource("jdkAndEclipse")
+	void integerIncrement(JavaCompiler jc) {
+		LambdaFactory factory = LambdaFactory.get(LambdaFactoryConfiguration.get().withJavaCompiler(jc));
+		Function<Integer, Integer> lambda = factory.createLambdaUnchecked("i -> i+1",
+				new TypeReference<Function<Integer, Integer>>() {
+				});
+		assertTrue(1 == lambda.apply(0));
+	}
+
+	@ParameterizedTest
+	@MethodSource("jdkAndEclipse")
+	void integerMultiply(JavaCompiler jc) {
+		LambdaFactory factory = LambdaFactory.get(LambdaFactoryConfiguration.get().withJavaCompiler(jc));
+		IntBinaryOperator lambda = factory.createLambdaUnchecked("(a,b) -> a*b",
+				new TypeReference<IntBinaryOperator>() {
+				});
+		assertEquals(1 * 2 * 3 * 4, IntStream.range(1, 5).reduce(lambda).getAsInt());
+	}
+
+	@ParameterizedTest
+	@MethodSource("jdkAndEclipse")
+	void integerToString(JavaCompiler jc) {
+		LambdaFactory factory = LambdaFactory.get(LambdaFactoryConfiguration.get().withJavaCompiler(jc));
+		Function<Integer, String> lambda = factory.createLambdaUnchecked("i -> \"ABC\"+i+\"DEF\"",
+				new TypeReference<Function<Integer, String>>() {
+				});
+		assertEquals("ABC101DEF", lambda.apply(101));
+	}
+
+	@ParameterizedTest
+	@MethodSource("jdkAndEclipse")
+	void useImportToAddBigDecimals(JavaCompiler jc) {
+		LambdaFactory factory = LambdaFactory
+				.get(LambdaFactoryConfiguration.get().withImports(BigDecimal.class).withJavaCompiler(jc));
+		assertDoesNotThrow(() -> {
```

Comment:
What's the point of adding that assertion? I think the test will fail anyway if the exception is thrown out of it, won't it?

Reply:
`createLambda` declares that it can throw a `LambdaCreationException`. Rather than declaring `throws LambdaCreationException` on the test case we wanted to explicitly check that no exception gets thrown within the test.